From f2e27bf1698c775c081a93903f8d5473250692c8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:31 -0400
Subject: [tools] perf/completion: Don't dictate perf install location

Message-id: <1408462094-14194-103-git-send-email-jolsa@redhat.com>
Patchwork-id: 88077
O-Subject: [PATCH RHEL7.1 BZ1131394 102/325] perf completion: Don't dictate perf install location
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 1ba6e01782fd2a94481e18b91b363636f8171565
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Thu Jul 4 18:11:25 2013 +0530

description
===========
The statement

  have perf

limits the locations in which to look for the perf program.  Moreover,
it depends on the bash-completion package to be installed.  Replace it
with a call to `type perf`.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index 56e6a12..50540cf 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -19,7 +19,7 @@ __ltrim_colon_completions()
  fi
 }
 
-have perf &&
+type perf &>/dev/null &&
 _perf()
 {
  local cur prev cmd
-- 
1.7.1